services/brig/src/Brig/API/Public.hs

```diff
@@ -477,6 +478,21 @@ getMultiUserPrekeyBundleH zusr qualUserClients = do
     throwStd (errorToWai @'E.TooManyClients)
   API.claimMultiPrekeyBundles (ProtectedUser zusr) qualUserClients !>> clientError
 
+getMultiUserPrekeyBundleHV3 ::
+  Member (Concurrency 'Unsafe) r =>
+  UserId ->
+  Public.QualifiedUserClients ->
+  (Handler r) Public.QualifiedUserClientPrekeyMapV3
+getMultiUserPrekeyBundleHV3 zusr qualUserClients = do
+  maxSize <- fromIntegral . setMaxConvSize <$> view settings
+  let Sum (size :: Int) =
+        Map.foldMapWithKey
+          (\_ v -> Sum . Map.size $ v)
+          (Public.qualifiedUserClients qualUserClients)
+  when (size > maxSize) $
+    throwStd (errorToWai @'E.TooManyClients)
```

Comment:
Can you factor out this and then use it in both versions of the handler?

Reply:
Done and done